### DISCUSSION TITLE
Float datatype in PyVSC

### DISCUSSION THREAD

**Discussion by suchi2108** [fvutils/pyvsc#195] [Q&A]
Hi, I am writing coverage model for ADC using PyVSC coverage constraints. I want to declare float datatype for analog input signal. I couldn't find a way to model it. Any suggestion ?

**Comment by mballance** [comment]
Hi @suchi2108, you are correct that PyVSC currently only supports integral types for randomization and coverage. If you are modeling both randomization and coverage, I recommend modeling floating-point values as scaled integer values (ie randomization variables have a domain 100x of what is desired, and are divided by 100 to produce a floating-point number). If you are only modeling coverage, there may be some additional options.
For now, can you confirm whether you are modeling both randomization and coverage of floating-point values, or just coverage? If an example would be useful, let me know and I can put one together.

Thanks,
Matthew

**Reply by suchi2108** [reply]
Hi Mathew, thank you so much for quick reply. 

I am randomizing it using Python 'random' module.  So right now, I am focussed on modeling coverage only with PyVSC.  An example would be really nice. :)

Best,
Suruchi

**Reply by suchi2108** [reply]
Hi Matthew, can you please elaborate how can I model coverage of floating-point value ?

Thanks in advance.
